File: 2367-number-of-arithmetic-triplets/2367-number-of-arithmetic-triplets.java
class Solution {
    public int arithmeticTriplets(int[] nums, int diff) {
        int c = 0;
        
        for(int i = 0; i < nums.length-2; i++){
            for(int j = i+1; j < nums.length-1; j++){
                for(int k = j+1; k < nums.length; k++){
                    if(nums[j] - nums[i] == diff && nums[k] - nums[j] == diff)
                        ++c;
                }
            }
        }
        return c;
    }
}

File: 2367-number-of-arithmetic-triplets/2367-number-of-arithmetic-triplets.swift
class Solution {
    func arithmeticTriplets(_ nums: [Int], _ diff: Int) -> Int {
        var cnt = 0
        var i = 0
        
        while i < nums.count - 2 {
            var j = i + 1
            while j < nums.count - 1 {
                var k = j + 1 
                while k < nums.count {
                    if nums[k]-nums[j] == diff && nums[j]-nums[i] == diff { cnt += 1 }
                    k += 1
                }
                j += 1
            }
            i += 1
        }
        
        return cnt
    }
}